daemon/cmd/daemon_main.go

```diff
@@ -525,7 +525,7 @@ func init() {
 	flags.String(option.KubeProxyReplacementHealthzBindAddr, defaults.KubeProxyReplacementHealthzBindAddr, "The IP address with port for kube-proxy replacement health check server to serve on (set to '0.0.0.0:10256' for all IPv4 interfaces and '[::]:10256' for all IPv6 interfaces). Set empty to disable.")
 	option.BindEnv(option.KubeProxyReplacementHealthzBindAddr)
 
-	flags.Bool(option.EnableHostPort, true, fmt.Sprintf("Enable k8s hostPort mapping feature (requires enabling %s)", option.EnableNodePort))
+	flags.Bool(option.EnableHostPort, false, fmt.Sprintf("Enable k8s hostPort mapping feature (requires enabling %s)", option.EnableNodePort))
```

Comment:
Yeah, I think idea was that once `enable-node-port` gets enabled, we automatically have support for `enable-host-port` as well since it reuses the very same code. It's the same behavior with `enable-external-ips`. Is there any issue you ran into?

Reply:
I'm unable to find the thread (Slack archived it) where I had provided more context, and we had agreed to change the default. Essentially, the semantics are confusing because `hostPort` config also needs `nodePort` to be enabled. If you don't read the note in the agent daemon source code, then you may be under the impression that `hostPort` is enabled by default, and the functionality should work.

Reply:
But changing the default will break setups as users would rely on this by now. How is this different to e.g. externalIPs which does the same? Does the confusion come from the `--help` output of the agent or the cmdref? Or is there anything missing in the kube-proxy free guide? Maybe another option could be to hide these knobs for cmdref and only have them documented in the kube-proxy free guide for expert users? (cc @brb )

Reply:
Yeah, I'm afraid that it will break `--kpr=partial` deployments which didn't explicitly specify `--enable-host-port=true`.